### ISSUE TITLE
how to change table name?

### ISSUE THREAD

**Issue by YvanYeo** [dfahlander/Dexie.js#713]
```
const db = new Dexie('mydb');
db.version(1).stores({
  friends: '++id, name, age'
});
db.version(2).stores({
  friends2: 'id, name, age'
}).upgrade(()=>{
  return db.friends.toArray().then(objs => {
    return db.friends2.bulkAdd(objs);
  });
});
db.version(3).stores({
  friends: null
});
```
I tried this, but it doesn't work, with an error message following:

```TypeError Cannot read property 'toArray' of undefined```

**Comment by dfahlander** [comment]
This is related to #105. It's still a bug.

Looking into it. I'll see what I could do...

**Comment by dfahlander** [comment]
This has been a headache for a long time (#88, #105 and maybe some other open issues), but I just submitted a solution that can resolve it. There were more visions for this, that is not taken into account. I really only needed to fix this particular lack of possibility to remove tables while still having access to them in the upgraders.

@YvanYeo Once my PR has passed all integration tests and published to npm (probably version 3.0.0-alpha.3), you can install it using

```
npm i dexie@latest
```

Then your upgrader code will need to use the transaction instance instead of db:

```js
const db = new Dexie('mydb');
db.version(1).stores({
  friends: '++id, name, age'
});
db.version(2).stores({
  friends2: 'id, name, age'
}).upgrade(tx =>{
  return tx.friends.toArray().then(objs => {
    return tx.friends2.bulkAdd(objs);
  });
});
db.version(3).stores({
  friends: null
});
```

There is a test that verifies this issue is resolved. You can also look at that code sample in https://github.com/dfahlander/Dexie.js/blob/issue713/test/tests-upgrading.js#L515

With this PR, it will also be possible to change primary key, as requested in #88 even though a bit verbosely. And the code will not work on Internet Explorer 10 or 11. https://github.com/dfahlander/Dexie.js/blob/issue713/test/tests-upgrading.js#L550
